scripts/install.sh

```diff
@@ -0,0 +1,112 @@
+
+# Define text formatting
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[1;34m'
+BOLD='\033[1m'
+NORMAL='\033[0m'
+
+# Function for logging with timestamp
+log() {
+    local LEVEL="$1"
+    shift
+    local MESSAGE="$*"
+    local TIMESTAMP
+    TIMESTAMP=$(date +"%Y-%m-%d %H:%M:%S")
+    echo -e "${TIMESTAMP} ${LEVEL} ${MESSAGE}"
+}
+
+# Logging helpers
+info_message() {
+    log "${BLUE}${BOLD}[INFO]${NORMAL}" "$*"
+}
+
+warn_message() {
+    log "${YELLOW}${BOLD}[WARNING]${NORMAL}" "$*"
+}
+
+error_message() {
+    log "${RED}${BOLD}[ERROR]${NORMAL}" "$*"
+}
+
+success_message() {
+    log "${GREEN}${BOLD}[SUCCESS]${NORMAL}" "$*"
+}
+
+print_step() {
+    log "${BLUE}${BOLD}[STEP]${NORMAL}" "$1: $2"
+}
+
+# Exit script with an error message
+error_exit() {
+    error_message "$1"
+    exit 1
+}
+
+# Check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Ensure root privileges, either directly or through sudo
+maybe_sudo() {
+    if [ "$(id -u)" -ne 0 ]; then
+        if command_exists sudo; then
+            sudo "$@"
+        else
+            error_message "This script requires root privileges. Please run with sudo or as root."
+            exit 1
+        fi
+    else
+        "$@"
+    fi
+}
+
+# Determine the OS and architecture
+case "$(uname)" in
+    "Linux") OS="unknown-linux-gnu"; BIN_DIR="/var/ossec/bin" ;;
+    "Darwin") OS="apple-darwin"; BIN_DIR="/Library/Ossec/bin" ;;
+    *) error_exit "Unsupported operating system: $(uname)" ;;
+esac
+
+ARCH=$(uname -m)
+case "$ARCH" in
+    "x86_64") ARCH="x86_64" ;;
+    "arm64"|"aarch64") ARCH="aarch64" ;;
+    *) error_exit "Unsupported architecture: $ARCH" ;;
+esac
+
+# Construct binary name and URL for download
+BIN_NAME="$APP_NAME-${ARCH}-${OS}"
+BASE_URL="https://github.com/ADORSYS-GIS/wazuh-cert-oauth2/releases/download/v$WOPS_VERSION"
+URL="$BASE_URL/$BIN_NAME"
+
+# Create a temporary directory and ensure it is cleaned up
+TEMP_DIR=$(mktemp -d) || error_exit "Failed to create temporary directory"
+trap 'rm -rf "$TEMP_DIR"' EXIT
+
+# Step 1: Download the binary file
+print_step 1 "Downloading $BIN_NAME from $URL..."
+curl -SL --progress-bar -o "$TEMP_DIR/$BIN_NAME" "$URL" || error_exit "Failed to download $BIN_NAME"
+
+# Step 2: Install the binary
+print_step 2 "Installing binary to $BIN_DIR..."
+maybe_sudo mkdir -p "$BIN_DIR" || error_exit "Failed to create directory $BIN_DIR"
+maybe_sudo mv "$TEMP_DIR/$BIN_NAME" "$BIN_DIR/$APP_NAME" || error_exit "Failed to move binary to $BIN_DIR"
+maybe_sudo chmod 750 "$BIN_DIR/$APP_NAME" || error_exit "Failed to set executable permissions on the binary"
+
+success_message "Installation and configuration complete! You can now use '$BIN_DIR/$APP_NAME' from your terminal."
+info_message "Run \n\n\t${GREEN}${BOLD}sudo $BIN_DIR/$APP_NAME o-auth2${NORMAL}\n\n to start configuring. If you don't have sudo on your machine, you can run the command without sudo."
```

Comment:
Here I can see the steps are 
1. Download the bin
2. Install the bin.

How's the user running it? Is he running it manually? Or is it starting by itself?